lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -158,6 +158,26 @@ function dosomething_northstar_get_northstar_user($drupal_id) {
 }
 
 /**
+ * Checks if a Northstar user exists for a given index & type.
+ * @param  string $index_type The type of index to query for (Email, Facebook, Mobile, _id)
+ * @param  string $index The index to query for
+ * @return bool|object
```

Comment:
Oh hmm, this feels iffy to me. Can we return either `object|null` or just `bool`? Not sure if the method changed from my first read-through, but it seems like it does different things rather unexpectedly (and so it isn't really a `exists_by_index` _or_ a `get_user_by_index`... it's a bit of both!)